service/src/main/java/com/epam/esm/service/service/certificate/CertificateServiceImpl.java

```diff
@@ -224,43 +269,12 @@ private void validateParameterValues(List<String> parameterValues) throws BadReq
         }
     }
 
-    private Certificate findOldCertificate(int certificateId) throws BadRequestException {
-        if (certificateId <= 0) {
-            throw new BadRequestException("Wrong certificate id!", ErrorCodes.CERTIFICATE_BAD_REQUEST);
-        }
-        return dao.getById(certificateId);
-    }
-
-    @Transactional
-    @Override
-    public Certificate patch(Certificate certificate)
-            throws ValidationException, BadRequestException, NotFoundException {
-        Certificate oldCertificate = findOldCertificate(certificate.getId());
-
-        if (oldCertificate == null) {
-            throw new BadRequestException("Wrong certificate id!", ErrorCodes.CERTIFICATE_BAD_REQUEST);
-        }
-        return update(updateCertificate(certificate, oldCertificate));
-    }
-
-    private Certificate updateCertificate(Certificate newCertificate, Certificate oldCertificate) {
-        if (newCertificate.getPrice() != null
-                && newCertificate.getPrice().compareTo(BigDecimal.ZERO) > 0) {
-            oldCertificate.setPrice(newCertificate.getPrice());
-        }
-        if (newCertificate.getDescription() != null) {
-            oldCertificate.setDescription(newCertificate.getDescription());
-        }
-        if (newCertificate.getDuration() != 0) {
-            oldCertificate.setDuration(newCertificate.getDuration());
-        }
-        if (newCertificate.getName() != null) {
-            oldCertificate.setName(newCertificate.getName());
-        }
-        if (CollectionUtils.isNotEmpty(newCertificate.getTags())) {
-            oldCertificate.setTags(newCertificate.getTags());
-        }
-        return oldCertificate;
+    private Certificate findOldCertificate(int certificateId)
```

Comment:
I don't think that this is a good name, since here we have logic that just getting any certificate by id, so no need to specify that this is an old one

Reply:
We also need to check if it is present. I tried to move logic into methods using it and get smth like : 
Certificate oldCertificate = dao.findById(certificate.getId()).orElseThrow(
                () -> new NotFoundException(String.format(notFoundErrorMessage, "id",
                        Integer.toString(certificate.getId())),
                        ErrorCodes.CERTIFICATE_NOT_FOUND));
                        
As I see  two occurencies of these need to be moved to separated method. Of course, if you insist, I won't do it

Reply:
I just meant that it's better if you rename it, as this logic doesn't know that this is the old certificate, it could be any other

Reply:
discussed